lld/COFF/Chunks.cpp

```diff
@@ -1147,4 +1148,81 @@ uint32_t ImportThunkChunkARM64EC::extendRanges() {
   return sizeof(arm64Thunk) - sizeof(uint32_t);
 }
 
+size_t Arm64XDynamicRelocEntry::getSize() const {
+  switch (type) {
+  case IMAGE_DVRT_ARM64X_FIXUP_TYPE_VALUE:
+    return sizeof(uint16_t) + size; // A header and a payload.
+  case IMAGE_DVRT_ARM64X_FIXUP_TYPE_DELTA:
+  case IMAGE_DVRT_ARM64X_FIXUP_TYPE_ZEROFILL:
+    llvm_unreachable("unsupported type");
+  }
+}
```

Comment:
This causes warnings when building with GCC:
```
../../lld/COFF/Chunks.cpp: In member function ‘size_t lld::coff::Arm64XDynamicRelocEntry::getSize() const’:
../../lld/COFF/Chunks.cpp:1177:1: warning: control reaches end of non-void function [-Wreturn-type]
 1177 | }
      | ^
```

This case is mentioned in the LLVM coding style guide, see https://llvm.org/docs/CodingStandards.html#don-t-use-default-labels-in-fully-covered-switches-over-enumerations - the default action to silence this warning with GCC is to add an `llvm_unreachable("")` after the switch; see e.g. b21de9b38f4b5f2ae6d49f973b683f118b9d58cb. (In this case, it feels a bit repetetive when some cases already end with `llvm_unreachable()` too, but they're different cases.)

Reply:
I created #122382 for a fix, sorry about that. The current `llvm_unreachable()` will be removed one ARM64X support is more complete. Those two reloc types are needed for the shared IAT.